examples/tutorials/advanced/focal_mechanisms.py

```diff
@@ -0,0 +1,449 @@
+  *strike*, *dip*, *rake*, *magnitude*
+- ``"gcmt"`` - global CMT:
+  *strike1*, *dip1*, *rake1*, *strike2*, *dip2*, *rake2*, *mantissa*,
+  *exponent*
+- ``"mt"`` - seismic moment tensor:
+  *mrr*, *mtt*, *mff*, *mrt*, *mrf*, *mtf*, *exponent*
+- ``"partial"`` - partial focal mechanism:
+  *strike1*, *dip1*, *strike2*, *fault_type*, *magnitude*
+- ``"principal_axis"`` - principal axis:
+  *t_value*, *t_azimuth*, *t_plunge*, *n_value*, *n_azimuth*, *n_plunge*,
+  *p_value*, *p_azimuth*, *p_plunge*, *exponent*
+
+The general structur for the input data is:
+
+-  xxx
+
+Please refer also the documentation on how to set up the input data in respect
+to the chosen input type and convention.
+
+This tutorial shows how to adjust the display of the beachballs:
+
+- Adjust outline
+- Fill quadrants with colors and patterns
+- Highlight nodal plane
+- Offset from event location
+- Size-coding and color-coding
+- Add label
+"""
+
+# sphinx_gallery_thumbnail_number = 7
+
+
+# Import the required packages
+import pygmt
+
+###############################################################################
+# Set up input data
+# -----------------
+# TODO - consistent with lists in introduction
+
+# Store focal mechanism parameters
+# in a 1-D array
+fm_sinlge = "xxx"
+# in pandas DataFrame
+fm_single = "xxx"
+# in a dictionary based on the Aki & Richards convention
+fm_single = dict(strike=166, dip=80, rake=74, magnitude=5.8)
+
+# Define study area: lon_min, lon_max, lat_min, lat_max
+# in degrees East or North
+study_area = [-84, -76, -4, 4]
+
+###############################################################################
+# Plot a single beachball
+# -----------------------
+# Required parameters are ``spec``, ``scale``, ``longitude`` and ``latitude``
+# (event location)
+
+fig = pygmt.Figure()
+
+# Create basic map of study area
+fig.coast(
+    region=study_area,
+    projection="M10c",
+    land="lightgray",
+    water="lightblue",
+    shorelines="1/0.5p,darkgray",
+    borders="1/0.5p,darkred",
+    frame=True,
+)
+
+# Pass the focal mechanism data through the spec parameter
+# additionally scale and event location are required
+fig.meca(
+    spec=fm_single,
+    scale="1c",  # in centimeters
+    longitude=-79.611,
+    latitude=0.904,
+)
+
+fig.show()
+
+
+###############################################################################
+# Adjust outline
+# --------------
+# parameters ``pen`` and ``outline``
+
+# Create new figure instance
+fig = pygmt.Figure()
+
+# Create basic map of study area
+fig.coast(
+    region=study_area,
+    projection="M10c",  # Mercator projection with width 10 centimeters
+    land="lightgray",
+    water="lightblue",
+    shorelines="1/0.5p,darkgray",
+    borders="1/0.5p,darkred",
+    frame=["af", "WSne"],
+)
+
+# Plot focal mechanism
+fig.meca(
+    spec=fm_single,
+    scale="1c",  # in centimeters
+    longitude=-79.611,  # event longitude
+    latitude=0.904,  # event latitude
+    pen="1p,blue,solid",
```

Comment:
```suggestion
    pen="1p,gold,solid",
```
Would suggest to use another (lighter) color than default value, makes it easier to distinguish the outline from the black fill.

Reply:
Thanks @michaelgrund for looking at this PR at this early stage. I will consider your review when working on this tutorial again. However, I will not be able to finish it in the next few days to get it into the v.0.10.0 release.